protocol/validator.go

```diff
@@ -0,0 +1,150 @@
+package protocol
+
+import (
+	"embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/santhosh-tekuri/jsonschema/v5"
+)
+
+// DataType represents the type of data being validated e.g. resource or message
+type DataType = string
```

Comment:
This should probably be a `type DataType string`. The difference is that `=` creates an alias, while this form creates a new type that is a string, which is probably what you want here.

Reply:
great catch @alecthomas , i actually initially went with `type DataType string` but ended up making it an alias in order to avoid having to type cast e.g. [here](https://github.com/TBD54566975/tbdex-go/blob/validation/protocol/validator.go#L93) and [here](https://github.com/TBD54566975/tbdex-go/blob/validation/protocol/validator.go#L47-L48)

and thankfully even with aliasing, vscode (with or without copilot) suggests the appropriate args which was the original motivation behind introducing `DataType`

![image](https://github.com/TBD54566975/tbdex-go/assets/4887440/94642b48-ca53-467f-aeb2-769be999026e)

lmk if you can think of any corners i may be painting myself into

Reply:
The corner is that there's zero additional type safety, but it's your call. It's rarely idiomatic to use `=`.

Reply:
very compelling point 🤣 and very smelly corner. changed from alias to new type https://github.com/TBD54566975/tbdex-go/commit/0ea9b354e3cff891e20db140016f68c6d8b57fb7

thanks @alecthomas